src/app/shared/store/effects.ts

```diff
@@ -0,0 +1,119 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { Router } from '@angular/router';
+
+import { AuthService } from '../services/auth.service';
+import { HttpService } from '../services/http.service';
+import {
+  setLogin,
+  setCharts,
+  setChartsSuccess,
+  setLoginFail,
+  setLoginSuccess,
+  setReports,
+  setReportsFail,
+  setReportsSuccess,
+  setUsers,
+  setUsersFail,
+  setUsersSuccess,
+  setChartsFail,
+} from './actions';
+import {
+  catchError,
+  concatAll,
+  exhaustMap,
+  map,
+  mergeMap,
+  of,
+  reduce,
+  withLatestFrom,
+} from 'rxjs';
+import { IAuthState, IPreparedChart } from '../models/interfaces';
+import { Store } from '@ngrx/store';
+import { AuthSelectors } from './selectors';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthorisationEffects {
+  constructor(
+    private actions: Actions,
+    private auth: AuthService,
+    private http: HttpService,
+    private router: Router,
+    private store: Store<IAuthState>
+  ) {}

```

Comment:
in effects we have variables for every effect, so constructor is going under all of them